src/AbstractBaseMenuElement.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+namespace RebelCode\WordPress\Admin\Menu;
+
+use ArrayIterator;
+use Dhii\Validation\Exception\ValidationFailedException;
+use RecursiveArrayIterator;
+use RecursiveIterator;
+
+/**
+ * Base functionality for a menu element.
+ *
+ * @since [*next-version*]
+ */
+abstract class AbstractBaseMenuElement extends AbstractMenuElement implements
+    MenuElementInterface,
+    RecursiveIterator
+{
+    /**
+     * Description.
+     *
+     * @since [*next-version*]
+     *
+     * @var ArrayIterator
+     */
+    protected $iterator;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function getKey()
+    {
+        return $this->_getKey();
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function getValue()
+    {
+        return $this->_getValue();
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function getChildren()
+    {
+        return new RecursiveArrayIterator($this->_getChildren());
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function hasChildren()
+    {
+        // What to do here? ChildrenAware::hasChildren() or RecursiveIterator::hasChildren() ?
```

Comment:
Yes, we have a problem. I also faced this problem with my `ConfigInterface` implementation. I have an idea of how to solve this using proper SoC. Let's talk about that ASAP.